qcodes_contrib_drivers/drivers/Keysight/Keysight_E36313A.py

```diff
@@ -0,0 +1,111 @@
+            attached.
+            name (str): The name of the channel
+            channum (int): The number of the channel in question (1-2)
+        """
+        # Sanity Check inputs
+        if name not in ['ch1', 'ch2', 'ch3']:
+            raise ValueError("Invalid Channel: {}, expected 'ch1' or 'ch2' or 'ch3'"
+                             .format(name))
+        if chan not in [1, 2, 3]:
+            raise ValueError("Invalid Channel: {}, expected '1' or '2' or '3'"
+                             .format(chan))
+
+        super().__init__(parent, name)
+
+        self.add_parameter('source_voltage',
+                           label="Channel {} Voltage".format(chan),
+                           get_cmd='SOURCE{:d}:VOLT?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SOURCE{:d}:VOLT {{:.8G}}'.format(chan),
+                           unit='V')
+
+        self.add_parameter('source_current',
+                           label="Channel {} Current".format(chan),
+                           get_cmd='SOURCE{:d}:CURR?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SOURCE{:d}:CURR {{:.8G}}'.format(chan),
+                           unit='A')
+
+        self.add_parameter('voltage',
+                           get_cmd='MEAS:VOLT? (@{:d})'.format(chan),
+                           get_parser=float,
+                           label='Channel {} Voltage'.format(chan),
+                           unit='V')
+
+        self.add_parameter('current',
+                           get_cmd='MEAS:CURR? (@{:d})'.format(chan),
+                           get_parser=float,
+                           label='Channel {} Current'.format(chan),
+                           unit='A')
+
+        self.add_parameter('resistance',
+                           get_cmd='MEAS:RES? (@{:d})'.format(chan),
+                           get_parser=float,
+                           label='Channel {} Resistance'.format(chan),
+                           unit='ohm')
+
+        self.add_parameter('voltage_limit',
+                           get_cmd='SENS{:d}:VOLT:PROT?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SENS{:d}:VOLT:PROT {{:.8G}}'.format(chan),
+                           label='Channel {} Voltage Limit'.format(chan),
+                           unit='V')
+
+        self.add_parameter('current_limit',
+                           get_cmd='SENS{:d}:CURR:PROT?'.format(chan),
+                           get_parser=float,
+                           set_cmd='SENS{:d}:CURR:PROT {{:.8G}}'.format(chan),
+                           label='Channel {} Current Limit',
+                           unit='A')
+
+        self.add_parameter('enable',
+                           get_cmd='OUTP{:d}?'.format(chan),
+                           set_cmd='OUTP{:d} {{:d}}'.format(chan),
```

Comment:
pg 71 of the manual https://literature.cdn.keysight.com/litweb/pdf/E36311-90008.pdf
suggest that SCPI for the enable/disable is 
OUTPut[:STATe]?[(@<chanlist>)]
OUTPut[:STATe]ON| 1 |OFF| 0, (@<chanlist>)] OUTPut[:STATe]?[(@<chanlist>)]

your code should look like:
get_cmd='OUTP? (@{:d})'.format(chan)
set_cmd='OUTP{:d} (@{:d})'.format(chan)